tests/waku_filter_v2/test_waku_client.nim

```diff
@@ -9,720 +9,941 @@ import
+  ../../../waku/node/peer_manager,
+  ../../../waku/waku_filter_v2,
+  ../../../waku/waku_filter_v2/client,
+  ../../../waku/waku_filter_v2/subscriptions,
+  ../../../waku/waku_core,
+  ../testlib/common,
   ../testlib/wakucore,
   ../testlib/testasync,
+  ../testlib/testutils,
   ./waku_filter_utils.nim
 
 let FUTURE_TIMEOUT = 1.seconds
 
 proc newPushHandlerFuture(): Future[(string, WakuMessage)] =
     newFuture[(string, WakuMessage)]()
 
+
 suite "Waku Filter - End to End":
-  var serverSwitch {.threadvar.}: Switch
-  var clientSwitch {.threadvar.}: Switch
-  var wakuFilter {.threadvar.}: WakuFilter
-  var wakuFilterClient {.threadvar.}: WakuFilterClient
-  var serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
-  var pubsubTopic {.threadvar.}: PubsubTopic
-  var contentTopic {.threadvar.}: ContentTopic
-  var contentTopicSeq {.threadvar.}: seq[ContentTopic]
-  var clientPeerId {.threadvar.}: PeerId
-  var pushHandlerFuture {.threadvar.}: Future[(string, WakuMessage)]
-
-  asyncSetup:
-    pushHandlerFuture = newPushHandlerFuture()
-    let messagePushHandler: FilterPushHandler = proc(
-      pubsubTopic: PubsubTopic, message: WakuMessage
-    ) {.async, gcsafe, closure.} =
-      pushHandlerFuture.complete((pubsubTopic, message))
-
-    pubsubTopic = DefaultPubsubTopic
-    contentTopic = DefaultContentTopic
-    contentTopicSeq = @[DefaultContentTopic]
-    serverSwitch = newStandardSwitch()
-    clientSwitch = newStandardSwitch()
-    wakuFilter = await newTestWakuFilter(serverSwitch)
-    wakuFilterClient = await newTestWakuFilterClient(clientSwitch)
-    wakuFilterClient.registerPushHandler(messagePushHandler)
-    
-    await allFutures(serverSwitch.start(), clientSwitch.start())
-    serverRemotePeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
-    clientPeerId = clientSwitch.peerInfo.toRemotePeerInfo().peerId
-  
-  asyncTeardown:
-    await allFutures(wakuFilter.stop(), wakuFilterClient.stop(), serverSwitch.stop(), clientSwitch.stop())
-
-  suite "Subscriber Ping":
-    asyncTest "Active Subscription Identification":
-      # Given
-      let 
-        subscribeResponse = await wakuFilterClient.subscribe(
+  suite "MessagePushHandler - Void":
+    var serverSwitch {.threadvar.}: Switch
+    var clientSwitch {.threadvar.}: Switch
+    var wakuFilter {.threadvar.}: WakuFilter
+    var wakuFilterClient {.threadvar.}: WakuFilterClient
+    var serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
+    var pubsubTopic {.threadvar.}: PubsubTopic
+    var contentTopic {.threadvar.}: ContentTopic
+    var contentTopicSeq {.threadvar.}: seq[ContentTopic]
+    var clientPeerId {.threadvar.}: PeerId
+    var messagePushHandler {.threadvar.}: FilterPushHandler
+    var pushHandlerFuture {.threadvar.}: Future[(string, WakuMessage)]
+
+    asyncSetup:
+      pushHandlerFuture = newPushHandlerFuture()
+      messagePushHandler = proc(
+        pubsubTopic: PubsubTopic, message: WakuMessage
+      ): Future[void] {.async, closure, gcsafe.} =
```

Comment:
Small nitpick:
I think is not needed to use `Future[void]`. This is done implicitly by the `async` pragma.

Reply:
Good to know, thanks!